### ISSUE TITLE
History Graphs not loading

### ISSUE THREAD

**Issue by Syralist** [home-assistant/frontend#9373]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since the update from 2021.5.5 to 2021.6.0 my history graphs are no longer loading. See the screenshot below, those are sensor cards in a horizontal stack but the custom mini-graph-card is also affected.

![image](https://user-images.githubusercontent.com/4690456/120891567-91ea7880-c609-11eb-999f-f402432e9491.png)

### Describe the behavior you expected

I except the graphs to load properly.

### Steps to reproduce the issue

I did only the update from 2021.5.5 to 2021.6.0

### What version of Home Assistant Core has the issue?

core-2021.6.2

### What was the last working version of Home Assistant Core?

core-2021.5.5

### In which browser are you experiencing the issue with?

Google Chrome 91.0.4472.77 (Offizieller Build) (64-Bit)

### Which operating system are you using to run this browser?

Windows 10

### Problem-relevant frontend configuration

```yaml
entity: sensor.tankerkoenig_aral_tankstelle_e10
graph: line
name: Aral
type: sensor
```

### Javascript errors shown in your browser console/inspector

```txt
I get 404 errors when the frontend tries to load the history:

Name: https://mydomain.duckdns.org:8123/api/history/period/2021-06-04T12:29:44.889Z?filter_entity_id=sensor.tankerkoenig_aral_tankstelle_e10&end_time=2021-06-05T12:29:44.889Z&minimal_response
Initiator: app.bcb38ac1.js, service_worker.js, chunk.9d638771aa57c2274890.js
```

### Additional information

I use the MariaDB Addon for the recorder. During the update the DB schema upgraded from 14 to 15 to 16 without error. There are no recorder or DB related errors in my log. 
I tried uninstalling and reinstalling the Addon. I reverted the DB to a snapshot from before the update. I even reconfigured the recorder to not use the MariaDB but the standard SQLite DB. Nothing changes the behaviour.

**Comment by spacegaier** [comment]
Does the API call work if you try it directly locally, to rule out any duckdns issues?

**Comment by theetron** [comment]
I also have problems with the state history. It loads, but it can take up to minute to show. Not from MariaDB but just a normal sensor and a sensor card. In the GUI it is slow and clicking on the graph to show the red lined graph is also slow.

**Comment by Syralist** [comment]
> Does the API call work if you try it directly locally, to rule out any duckdns issues?

On my pc i have the local ip in my HOSTS file for the duck dns url. 
In the companion app i use the nabu casa url. 
Both show the same behavior.

**Comment by spacegaier** [comment]
Let me rephrase: I want to know if it is a frontend issue (so URL in itself is wrong, although it looks correct to me) or if the backend is the issue. That is why I asked if the URL locally (e.g. on the machine where HA runs) works or not.

Since it is a 404 (and not a 401 that I would expect if you were to try to call the URL locally without login credentials) it sounds more like a network or backend issue to me.

**Comment by spacegaier** [comment]
> 
> 
> I also have problems with the state history. It loads, but it can take up to minute to show. Not from MariaDB but just a normal sensor and a sensor card. In the GUI it is slow and clicking on the graph to show the red lined graph is also slow.

Then best to create a ticket for it (if there isn't one already) and not hijack this one. Since you state the graphs do load for you, it is clearly different from this one here 😉.

**Comment by Syralist** [comment]
> Let me rephrase: I want to know if it is a frontend issue (so URL in itself is wrong, although it looks correct to me) or if the backend is the issue. That is why I asked if the URL locally (e.g. on the machine where HA runs) works or not.
> 
> Since it is a 404 (and not a 401 that I would expect if you were to try to call the URL locally without login credentials) it sounds more like a network or backend issue to me.

The machine where HA runs is a Raspberry with HaOS. How do I run the API call locally?

**Comment by spacegaier** [comment]
Either really locally in the console of the RPi (wget or curl), although I never tried that or from a computer in the same network using the direct IP of the RPi rather than going through duckdns.

**Comment by Syralist** [comment]
It's the same when I use the IP directly.

**Comment by Syralist** [comment]
my graphs are still not working even with the lastest Core version

**Comment by ludeeus** [comment]
Looks like your user is not an administrator, verify your access by pasting `window.document.querySelector("home-assistant").hass.user` in the browser console.

**Comment by Syralist** [comment]
yes my user is an admin

`window.document.querySelector("home-assistant").hass.user

{id: "a18d234a9d024c7fad65b971fe2d7734", name: "Thomas Helmke", is_owner: true, is_admin: true, credentials: Array(1), …}`

**Comment by ludeeus** [comment]
And you are sure that the history integration has properly loaded?

**Comment by Syralist** [comment]
Huh! the history integration was indeed not loaded! seems like I deliberately removed it at some point. now I feel stupid, thanks for pointing me in the right direction! :)

**Comment by ludeeus** [comment]
Glad it got sorted out 🎉
